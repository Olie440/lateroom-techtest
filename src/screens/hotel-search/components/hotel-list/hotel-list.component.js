import React from 'react';
import { connect } from 'react-redux';

// Ideally we would have an id for each hotel we could use as the key
export function HotelRow({ name, starRating, facilities }, index) {
    return (
        <div className="hotel-list__row" key={index}>
            <p className="hotel-list__title">{name}</p>
            <p>Rating: {starRating}</p>
            <p>Facilities: {facilities.join(', ')}</p>
        </div>
    );
}

export function HotelList({ hotels }) {
    if (hotels.state !== 'Success') {
        return null;
    }

    return (
        <div className="hotel-list">
            { hotels.data.map(HotelRow) }
        </div>
    );
}

export function mapStateToProps(state) {
    return {
        hotels: state.hotels
    }
}

export default connect(mapStateToProps)(HotelList);
