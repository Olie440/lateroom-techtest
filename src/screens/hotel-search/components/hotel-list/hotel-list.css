.hotel-list {
    background: #ffffff;
    border: 1px solid #cccccc;
    border-top: none;
}

.hotel-list__row {
    padding: 1em;
}

.hotel-list__row:nth-child(even) {
    background: #f0f0f0;
}

.hotel-list__title {
    font-size: 1.25rem;
    font-weight: bold;
    margin-bottom: 0.5rem;
}

.hotel-list__sub {
    font-size: 0.75rem;
    color: #777777;
    margin-bottom: 0.25rem;
}
