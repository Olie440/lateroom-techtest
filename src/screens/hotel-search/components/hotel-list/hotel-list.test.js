import React from 'react';
import { HotelList, mapStateToProps } from './hotel-list.component';
import mockHotels from '../../../../../__mocks__/mock-hotels';
import { shallow } from 'enzyme';

describe('HotelList', () => {
    it('renders the hotels when state = "Success"', () => {
        const props = {
            hotels: {
                state: 'Success',
                data: mockHotels()
            }
        };

        const component = shallow(<HotelList {...props} />);
        expect(component).toMatchSnapshot();
    });

    it('renders the loading component when state = "Loading"', () => {
        const props = {
            hotels: {
                state: 'Loading',
                data: null
            }
        };

        const component = shallow(<HotelList {...props} />);
        expect(component).toMatchSnapshot();
    });
})

describe('mapStateToProps', () => {
    it('only returns the hotels key from the state', () => {
        const testStore = { foo: 'bar', hotels: mockHotels() };
        const result = mapStateToProps(testStore);
        expect(result).toEqual({
            hotels: mockHotels()
        });
    });
});
