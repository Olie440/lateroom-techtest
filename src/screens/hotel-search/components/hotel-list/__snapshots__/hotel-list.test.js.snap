// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelList renders nothing when state = "None" 1`] = `""`;

exports[`HotelList renders the error component when state = "Error" 1`] = `
<div
  className="hotel-list"
>
  <Error
    message="Error loading Hotel Data, please refresh to try again."
  />
</div>
`;

exports[`HotelList renders the hotels when state = "Success" 1`] = `
<div
  className="hotel-list"
>
  <div
    className="hotel-list__row"
    key="0"
  >
    <p
      className="hotel-list__title"
    >
      hotelone
    </p>
    <p>
      Rating: 
      5
    </p>
    <p>
      Facilities: 
      car park, pool
    </p>
  </div>
  <div
    className="hotel-list__row"
    key="1"
  >
    <p
      className="hotel-list__title"
    >
      hoteltwo
    </p>
    <p>
      Rating: 
      3
    </p>
    <p>
      Facilities: 
      car park, gym
    </p>
  </div>
  <div
    className="hotel-list__row"
    key="2"
  >
    <p
      className="hotel-list__title"
    >
      hotelthree
    </p>
    <p>
      Rating: 
      3
    </p>
    <p>
      Facilities: 
    </p>
  </div>
</div>
`;

exports[`HotelList renders the loading component when state = "Loading" 1`] = `
<div
  className="hotel-list"
>
  <Loading
    message="Loading Hotel Data."
  />
</div>
`;
